src/axes/numericAxis.ts

```diff
@@ -29,21 +30,36 @@ export module Axes {
     }
 
     protected _computeWidth() {
+      var maxTextWidth = this._approximateTextWidth ? this._computeApproximateTextWidth() : this._computeExactTextWidth();
+
+      if (this._tickLabelPositioning === "center") {
+        this._computedWidth = this._maxLabelTickLength() + this.tickLabelPadding() + maxTextWidth;
+      } else {
+        this._computedWidth = Math.max(this._maxLabelTickLength(), this.tickLabelPadding() + maxTextWidth);
+      }
+
+      return this._computedWidth;
+    }
+
+    private _computeExactTextWidth(): number {
       var tickValues = this._getTickValues();
       var textLengths = tickValues.map((v: any) => {
         var formattedValue = this.formatter()(v);
         return this._measurer.measure(formattedValue).width;
       });
 
-      var maxTextLength = Utils.Math.max(textLengths, 0);
+      return Utils.Math.max(textLengths, 0);
+    }
 
-      if (this._tickLabelPositioning === "center") {
-        this._computedWidth = this._maxLabelTickLength() + this.tickLabelPadding() + maxTextLength;
-      } else {
-        this._computedWidth = Math.max(this._maxLabelTickLength(), this.tickLabelPadding() + maxTextLength);
-      }
+    private _computeApproximateTextWidth(): number {
+      var tickValues = this._getTickValues();
+      var mWidth = this._measurer.measure("M").width;
+      var textLengths = tickValues.map((v: any) => {
```

Comment:
I copied the above code since I don't know TS very well. I'll investigate the proper typing.